client/src/models/Position.js

```diff
@@ -63,7 +63,19 @@ export default class Position extends Model {
           org => org && org.uuid
         ),
       person: yup.object().nullable().default({}),
-      location: yup.object().nullable().default({})
+      location: yup
+        .object()
+        .nullable()
+        .default(null)
+        .label("Location")
+        .when("type", {
+          is: Position.TYPE.ADVISOR,
+          then: this.yupSchema.test(
+            "Field is required for advisor",
+            // eslint-disable-next-line no-template-curly-in-string
+            "${path} is required",
+            loc => loc && loc.uuid)
+        })
```

Comment:
You shouldn't refer `this.yupSchema` inside the schema; change this code to:
```
        .when("type", (type, schema) =>
          type === Position.TYPE.ADVISOR
            ? schema.required("Location is required for an advisor position")
            : schema.nullable()
        )
```

Reply:
Note that a location should also be required for `Position.TYPE.SUPER_USER` and `Position.TYPE.ADMINISTRATOR` (not yet added to the proposed change above).

Reply:
The `Position.TYPE.ADVISOR` does not indicate the current users privileges. In the case above, `type` is used to determine whether the position is going to be for a NATO Billet or Afghan Tashkil (according to the issue, only NATO Billet type should be effected by the change and Afghan Tashkil should remain as it is). The above schema works for all users. Since only super users and admins can create a position, they are both effected by the change therefore should we make additional checks if the current user is a super user or an admin?

Reply:
_Iff_ `[Position.TYPE.ADVISOR, Position.TYPE.SUPER_USER, Position.TYPE.ADMINISTRATOR].includes(type)` then the position is an advisor, else the position is a principal (i.e. `Position.TYPE.PRINCIPAL === type`). In all three cases, a location is required.

Reply:
So, for simplicity, the code could become:
```
        .when("type", (type, schema) =>
          type === Position.TYPE.PRINCIPAL
            ? schema.nullable()
            : schema.required(
              `Location is required for ${advisorPosition.name}`
            )
        )
```